src/charm.py

```diff
@@ -128,61 +108,79 @@ def _get_database_relation_info(self) -> dict:
             "endpoints": relation_data["endpoints"],
         }
 
-    def _update_container(self, event) -> None:
-        """Update configs, pebble layer and run database migration."""
+    def _run_sql_migration(self) -> None:
+        """Runs database migration."""
+        process = self._container.exec(
+            ["kratos", "migrate", "sql", "-e", "--config", self._config_file_path, "--yes"],
+        )
+        try:
+            stdout, _ = process.wait_output()
+            logger.info(f"Successfully executed automigration: {stdout}")
+        except ExecError as err:
+            logger.error(f"Exited with code {err.exit_code}. Stderr: {err.stderr}")
+            self.unit.status = BlockedStatus("Database migration job failed")
+
+    def _on_pebble_ready(self, event) -> None:
+        """Event Handler for pebble ready event."""
         if not self._container.can_connect():
             event.defer()
             logger.info("Cannot connect to Kratos container. Deferring event.")
             self.unit.status = WaitingStatus("Waiting to connect to Kratos container")
             return
 
-        if not self.model.relations["pg-database"]:
-            logger.error("Missing required relation with postgresql")
-            self.model.unit.status = BlockedStatus("Missing required relation with postgresql")
-            return
+        self.unit.status = MaintenanceStatus("Configuring/deploying resources")
+        with open("src/identity.default.schema.json", encoding="utf-8") as schema_file:
+            schema = schema_file.read()
+            self._container.push(self._identity_schema_file_path, schema, make_dirs=True)
+        self._container.add_layer(self._container_name, self._pebble_layer, combine=True)
+        logger.info("Pebble plan updated with new configuration, replanning")
+        self._container.replan()
+        self.unit.status = ActiveStatus()
 
-        if not self.database.is_database_created():
+    def _on_database_created(self, event) -> None:
```

Comment:
You observe `database_created` event, but run `_on_database_changed` method when it's emitted. Did you mean to run this one instead? It's not used anywhere else